libs/tree/tree_test.go

```diff
@@ -227,16 +266,36 @@ func TestGetNext(t *testing.T) {
 	}
 }
 
-func BenchmarkInsert(b *testing.B) {
-	tree := new(llrb)
+// BenchmarkLLRBInsert-8   	 1399514	       770 ns/op
+func BenchmarkLLRBInsert(b *testing.B) {
+	benchmarkInsert(b, NewLLRB)
+}
+
+// BenchmarkBTreeInsert-8   	 1000000	      1561 ns/op
+func BenchmarkBTreeInsert(b *testing.B) {
+	benchmarkInsert(b, NewBTree)
+}
+
+func benchmarkInsert(b *testing.B, treeGen func() BalancedTree) {
+	tree := treeGen()
 	for i := 0; i < b.N; i++ {
 		tree.Insert(NodeKey{Priority: uint64(i)}, getRandomBytes(1)[0])
 	}
 }
 
-func BenchmarkRemove(b *testing.B) {
+// BenchmarkLLRBRemove-8   	 1000000	      1382 ns/op
+func BenchmarkLLRBRemove(b *testing.B) {
+	benchmarkRemove(b, NewLLRB)
+}
+
+// BenchmarkBTreeRemove-8   	 1607142	       810 ns/op
+func BenchmarkBTreeRemove(b *testing.B) {
```

Comment:
benchmark on get next?

Reply:
What about add getNext benchmark on mempool level?

Reply:
doesn't hurt if we also test here